facilitator/src/manifest.rs

```diff
@@ -0,0 +1,336 @@
+}
+
+impl SpecificManifest {
+    fn from_https(base_path: &str, peer_name: &str) -> Result<SpecificManifest> {
+        if !base_path.starts_with("https://") {
+            return Err(anyhow!("Manifest must be fetched over HTTPS"));
+        }
+        let manifest_url = format!("{}/{}/specific-manifest.json", base_path, peer_name);
+        let response = ureq::get(&manifest_url)
+            // By default, ureq will wait forever to connect or
+            // read.
+            .timeout_connect(10_000) // ten seconds
+            .timeout_read(10_000) // ten seconds
+            .call();
+        if response.error() {
+            return Err(anyhow!("failed to fetch specific manifest: {:?}", response));
+        }
+        SpecificManifest::from_reader(response.into_reader())
+    }
+
+    fn from_file(path: &Path) -> Result<SpecificManifest> {
+        SpecificManifest::from_reader(File::open(path).context("failed to open manifest file")?)
+    }
+
+    fn from_reader<R: Read>(reader: R) -> Result<SpecificManifest> {
+        let manifest: SpecificManifest =
+            from_reader(reader).context("failed to decode JSON specific manifest")?;
+        if manifest.format != 0 {
+            return Err(anyhow!("unsupported manifest format {}", manifest.format));
+        }
+        Ok(manifest)
+    }
+
+    /// Returns the ECDSA P256 public key corresponding to the provided key
+    /// identifier, if it exists in the manifest.
+    fn batch_signing_public_key(&self, identifier: &str) -> Result<UnparsedPublicKey<Vec<u8>>> {
+        // No Rust crate that we have found gives us an easy way to parse PKIX
+        // SubjectPublicKeyInfo structures to get at the public key which can
+        // then be used in ring::signature. Since we know the keys we deal with
+        // should always be ECDSA P256, we can instead check that the binary
+        // blob inside the PEM has the expected prefix for this kind of key in
+        // this kind of encoding, as suggested in this GitHub issue on ring:
+        // https://github.com/briansmith/ring/issues/881
+        let key = self
+            .batch_signing_public_keys
+            .get(identifier)
+            .context(format!("no value for key {}", identifier))?;
+
+        let pem = pem::parse(&key.public_key)
+            .context(format!("failed to parse key entry {} as PEM", identifier))?;
+        if pem.tag != "PUBLIC KEY" {
+            return Err(anyhow!(
+                "key for identifier {} is not a PEM encoded public key"
+            ));
+        }
+        if pem.contents.len() < ECDSA_P256_SPKI_PREFIX.len() {
+            return Err(anyhow!("PEM contents not long enough to contain ASN.1 encoded ECDSA P256 SubjectPublicKeyInfo"));
+        }
+        if &pem.contents[..ECDSA_P256_SPKI_PREFIX.len()] != ECDSA_P256_SPKI_PREFIX {
+            return Err(anyhow!(
+                "PEM contents are not ASN.1 encoded ECDSA P256 SubjectPublicKeyInfo"
+            ));
+        }
```

Comment:
Suggestion: Use `split_at` which lets you avoid repeating the lenght a few times.
```rust
let (prefix, key) = pem.contents.split_at(ECDSA_P256_SPKI_PREFIX.len());
// Check if prefix != ECDSA_P256_SPKI_PREFIX
// Otherwise, go ahead and return key.
```

Unfortunately `strip_prefix` isn't stable yet otherwise that'd be an even better fit.

Reply:
I wanted to use `strip_prefix`, too. However `split_at` is _much_ nicer and removes a few slice operators, which makes me happy.